language/tools/move-unit-test/src/test_runner.rs

```diff
@@ -480,4 +508,169 @@ impl SharedTestingConfig {
 
         stats
     }
+
+    #[cfg(feature = "evm-backend")]
+    fn execute_via_evm(&self, yul_source: &str) -> (ExecuteResult, Duration) {
+        let (code, _) = evm_exec_utils::compile::solc_yul(yul_source, false).expect(
+            "Failed to compile yul source into EVM bytecode. This should not have happened.",
+        );
+
+        let vicinity = MemoryVicinity {
+            gas_price: 0.into(),
+            origin: H160::zero(),
+            chain_id: 0.into(),
+            block_hashes: vec![],
+            block_number: 0.into(),
+            block_coinbase: H160::zero(),
+            block_timestamp: 0.into(),
+            block_difficulty: 0.into(),
+            block_gas_limit: U256::MAX,
+            block_base_fee_per_gas: 0.into(),
+        };
+
+        let mut exec = Executor::new(&vicinity);
+
+        let now = Instant::now();
+        let res = exec.execute_custom_code(H160::zero(), H160::zero(), code, vec![]);
+        let elapsed = now.elapsed();
+
+        (res, elapsed)
+    }
+
+    #[cfg(feature = "evm-backend")]
+    fn exec_module_tests_evm(
+        &self,
+        test_plan: &ModuleTestPlan,
+        output: &TestOutput<impl Write>,
+    ) -> TestStatistics {
+        panic!();
+
+        let mut stats = TestStatistics::new();
+
+        let model = run_model_builder_with_options_and_compilation_flags(
+            &self.source_files,
+            &[],
+            ModelBuilderOptions::default(),
+            Flags::testing(),
+            self.named_address_values.clone(),
+        )
+        .unwrap_or_else(|e| panic!("Unable to build stackless bytecode: {}", e));
+
+        let gen_options = move_to_yul::options::Options::default();
+        for (function_name, test_info) in &test_plan.tests {
+            let yul_code = move_to_yul::generator::Generator::run_for_unit_test(
+                &gen_options,
+                &model,
+                &test_plan.module_id,
+                &IdentStr::new(function_name).unwrap(),
+            );
+
+            let (res, duration) = self.execute_via_evm(&yul_code);
+
+            let abort_code = || -> u64 {
+                assert!(res.return_value.len() == 8);
+
+                u64::from_be_bytes(res.return_value.as_slice().try_into().unwrap())
+            };
+
+            let test_run_info =
+                || -> TestRunInfo { TestRunInfo::new(function_name.to_string(), duration, 0) };
+
+            // TODO: gas/timeout
+            // TODO: arguments
+            // TODO: locations
+
+            match (test_info.expected_failure.as_ref(), &res.exit_reason) {
+                // Test expected to succeed or abort with a specific abort code, but ran into an internal error.
+                (None | Some(ExpectedFailure::ExpectedWithCode(_)), ExitReason::Revert(_))
+                    if abort_code() == !0u64 =>
```

Comment:
Is u64 max a differentiated (non-error) return value in the EVM?

Also a nit: May want to use `std::u64::MAX` here instead?

Reply:
> Is u64 max a differentiated (non-error) return value in the EVM?
Yes.

> Also a nit: May want to use std::u64::MAX here instead?
Will do.